### PR TITLE
Augment `get_route_for_model()` to support REST API routes.

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #DNE
# What's Changed
- This adds an `api=True` argument to `nautobot.utilities.utils.get_route_for_model()`
- Replaced almost all bespoke implementations of manual route construction for plugins or API routes with calls to `get_route_for_model()`
- Expanded the Development Best Practices guide with a section on "Getting URL Routes"

# TODO

- [ ] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/development/best-practices.md [hunk 1]
```diff
@@ -79,6 +79,71 @@ class ExampleModel(PrimaryModel):
...
+
+```python
+>>> get_route_for_model(ExampleModel, "list")
+"plugins:example_plugin:examplemodel_list"
+>>> get_route_for_model(ExampleModel, "list", api=True)
+"plugins-api:example_plugin-api:examplemodel-list"
+```
+
+!!! tip
+    The first argument may also optionally be an instance of a model, or may be substituted a with string using the dotted notation of `{app_label}:{model}` (e.g. `dcim.device`).
```
Comment (glennmatthews) [inline nautobot/docs/development/best-practices.md:145]:
```suggestion
    The first argument may also optionally be an instance of a model, or a string using the dotted notation of `{app_label}.{model}` (e.g. `dcim.device`).
```

File: nautobot/utilities/utils.py [hunk 2]
```diff
@@ -54,27 +54,40 @@ def csv_format(data):
...
+
     Examples:
         >>> get_route_for_model(Device, "list")
         "dcim:device_list"
+        >>> get_route_for_model(Device, "list", api=True)
+        "dcim-api:device-list"
+        >>> get_route_for_model(ExampleModel, "list")
+        "plugins:example_plugin:examplemodel_list"
+        >>> get_route_for_model(ExampleModel, "list", api=True)
+        "plugins-api:example_plugin-api:examplemodel-list"
```
Comment (glennmatthews) [inline nautobot/utilities/utils.py:78]:
```suggestion
        "plugins-api:example_plugin-api:examplemodel-list"
        >>> get_route_for_model("dcim.site", "list")
        "dcim:site_list"
        >>> get_route_for_model("dcim.site", "list", api=True)
        "dcim-api:site-list"
```

**Review by glennmatthews — COMMENTED** [review]
I think there's also a case in `nautobot.utilities.views.GetReturnURLMixin.get_return_url` that could perhaps also use `get_route_for_model`?

**Comment by glennmatthews** [comment]
@jathanism What's needed to move this out of draft status?

**Review by bryanculver — CHANGES_REQUESTED** [review]
Let's get this buttoned up and passing tests to simplify some other PRs.

If we miss a few of the old-style calls that's fine but getting the method available will help continued drift from say #2348 & #2371

File: nautobot/docs/development/best-practices.md [hunk 3]
```diff
@@ -79,6 +79,71 @@ class ExampleModel(PrimaryModel):
...
+## Getting URL Routes
+
+When developing new models a need often arises to retrieve a reversible route for a model to access it in either the web UI or the REST API. When this time comes, you **must** always use `nautobot.utilities.utils.get_route_for_model`. You **must not** write your own logic to construct route names.
+
+```python
+from nautobot.utilities.utils import get_route_for_model
+```
+
+This utility function supports both UI and API views for both Nautobot core apps and Nautobot plugins.
+
```
Comment (glennmatthews) [inline nautobot/docs/development/best-practices.md:91]: Can we add a note here that API view support was added in 1.4.3?

File: nautobot/docs/development/best-practices.md [hunk 4]
```diff
@@ -79,6 +79,71 @@ class ExampleModel(PrimaryModel):
...
+## UI Routes
```
Comment (glennmatthews) [inline nautobot/docs/development/best-practices.md:92]:
```suggestion
### UI Routes
```

**Review by bryanculver — APPROVED** [review]
:shipit:

**Pull request merged**